tests/kernel/build_git_kernel.pm

```diff
@@ -0,0 +1,54 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: build, install and boot a custom, upstream kernel tree
+#          from an arbitrary git tree
+#
+# Maintainer: Michael Moese <mmoese@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use power_action_utils 'power_action';
+
+sub run {
+    my $self = shift;
+    my $git_tree = get_required_var('KERNEL_GIT_TREE');
+    my $git_branch = get_var('KERNEL_GIT_BRANCH', 'master');
+
+    $self->select_serial_terminal;
+
+    # download, compile and install a kernel tree from git
+    zypper_call('in git-core ncurses-devel gcc flex bison libelf-devel libopenssl-devel');
+    assert_script_run("git clone $git_tree linux");
```

Comment:
Just `git clone --depth 1 --single-branch --branch $git_branch`?

Reply:
Not sure we don't want to test several branches without re-cloning in the future.. so getting everything may be the better choice?

Reply:
At that point it could be changed back trivially.

Reply:
ok let's have it that way